### ISSUE TITLE
[FEATURE REQ] Show report button on the Activity Log page

### ISSUE THREAD

**Issue by elvisduric** [microsoft/playwright-testing-service#47]
**Is your feature request related to a problem? Please describe.**
With this playwright service, we could run the test cases but the test report will still be on our local machine (or stored as a GitHub artefact, etc.). It would be helpful and beneficial to have the option to present test run reports on the Playwright Portal page.

**Describe the solution you'd like**
Currently, on the Playwright Portal, we can see each test run we started over the Playwright service. The small button or another indication for the report could be helpful and beneficial to the user to immediately see the results of that run.

AB#1890161

**Comment by ranat5** [comment]
+1
Use case - We would like to use Playwright testing as service offer. Currently, when we try to run the tests, we do see the run result under the Activity Log but displaying the results of the tests in the dashboard will help us immensely to visualise the test results (something similar to the process that we can visualise on our browsers locally when we run: npx playwright show-report)

**Comment by RazvanFilipescu** [comment]
+ 1

My use case would be having a comprehensive reporting tool embedded in the service. Some of the features that would make it great are:
- Having pre-defined and customizable graphical reports (pie charts, bar graphs, etc).
- Displaying test results, per test run. This should be in a graphical way (color-coded e.g. with red for failed and green for passed) as well as having a detailed log of the traces currently being displayed locally. 
- Displaying summary test results, for a number of test runs (e.g. maybe show all test runs and their results per time interval)
- Having the option of saving the Playwright *.zip trace files in an Azure Container e.g. and having these readily available beside the test result if you want to investigate further.
- Highlighting flaky tests per test run (if you have a retry mechanism enabled) and per a selection of test runs (where one test may fail in one run but be ok in another).
* (possibly unrelated but might make use of the historic test run information - reports) Having the option to orchestrate/optimize test runs based on previous test run data. An example would be starting the longest duration test on the first machine, the second longest on the second machine, etc. and, once all machines are used up, start the pending longest test on the first machine that becomes available.

**Comment by magesh-chandran** [comment]
+1
To start with, 
1. would like to have link to test results with meta data like release, browser, OS, date, ... 
2. Would like to have ability to Create my own reports (charts, Pie, table) using meta data as filters .. maybe something like powerBI

**Comment by chris-strudel** [comment]
+1  I'd take it even further.  I think the videos and traces should be available in the ADO pipeline run itself.  The integration should be seamless.  That's currently the major limitation of ADO in my opinion - test reporting.    Each test case run should have it's associated video and trace available.

**Comment by puagarwa** [comment]
@chris-strudel There is issue in ADO pipelines from quite some time with thier JUNIT file parsing which cause artifacts to not show up in ADO test tab. I myself have filed this issue in thier repo sometime but unfortunately there is no traction. https://github.com/microsoft/azure-pipelines-tasks/issues/16349

**Comment by puagarwa** [comment]
> +1 I'd take it even further. I think the videos and traces should be available in the ADO pipeline run itself. The integration should be seamless. That's currently the major limitation of ADO in my opinion - test reporting. Each test case run should have it's associated video and trace available for immediate viewing in ADO.
> 
> The documentation has the build step:
> 
> * task: PublishPipelineArtifact@1
>   displayName: Upload Playwright report
>   inputs:
>   targetPath: '$(MainProjectDirectory)/test-results' # update accordingly
>   artifact: 'Playwright tests'
>   publishLocation: 'pipeline'
> 
> If we're going to publish to the pipeline then publish to the pipeline all the way!

@elvisduric and @chris-strudel ADO team recently anounced the attachment support for JUNIT, We should get playwright artifacts with every test now in ADO https://devblogs.microsoft.com/devops/junit-attachments-support-for-publish-test-results/

**Comment by chris-strudel** [comment]
> > +1 I'd take it even further. I think the videos and traces should be available in the ADO pipeline run itself. The integration should be seamless. That's currently the major limitation of ADO in my opinion - test reporting. Each test case run should have it's associated video and trace available for immediate viewing in ADO.
> > The documentation has the build step:
> > 
> > * task: PublishPipelineArtifact@1
> >   displayName: Upload Playwright report
> >   inputs:
> >   targetPath: '$(MainProjectDirectory)/test-results' # update accordingly
> >   artifact: 'Playwright tests'
> >   publishLocation: 'pipeline'
> > 
> > If we're going to publish to the pipeline then publish to the pipeline all the way!
> 
> @elvisduric and @chris-strudel ADO team recently anounced the attachment support for JUNIT, We should get playwright artifacts with every test now in ADO https://devblogs.microsoft.com/devops/junit-attachments-support-for-publish-test-results/

Hi @puagarwa! Thank you for letting us know. I followed the link and set it up. That is a step in the right direction, but what we really need is to be able to view those traces and videos within the ADO tests tab. There needs to be some UX support. When you have 100+ tests and multiple failures, it just isn't practical to have to download each video one at a time and then launch it in a viewer. Also, at present, I don't think there is a way to consume the traces that are created. Those are usually shown in the HTML reporter. When support for those features is added, I think Playwright will surge forward in adoptability. As it stands right now, the reporting is still the weakest spot for Playwright with ADO.

**Comment by puagarwa** [comment]
> > > +1 I'd take it even further. I think the videos and traces should be available in the ADO pipeline run itself. The integration should be seamless. That's currently the major limitation of ADO in my opinion - test reporting. Each test case run should have it's associated video and trace available for immediate viewing in ADO.
> > > The documentation has the build step:
> > > 
> > > * task: PublishPipelineArtifact@1
> > >   displayName: Upload Playwright report
> > >   inputs:
> > >   targetPath: '$(MainProjectDirectory)/test-results' # update accordingly
> > >   artifact: 'Playwright tests'
> > >   publishLocation: 'pipeline'
> > > 
> > > If we're going to publish to the pipeline then publish to the pipeline all the way!
> > 
> > 
> > @elvisduric and @chris-strudel ADO team recently anounced the attachment support for JUNIT, We should get playwright artifacts with every test now in ADO https://devblogs.microsoft.com/devops/junit-attachments-support-for-publish-test-results/
> 
> Hi @puagarwa! Thank you for letting us know. I followed the link and set it up. That is a step in the right direction, but what we really need is to be able to view those traces and videos within the ADO tests tab. There needs to be some UX support. When you have 100+ tests and multiple failures, it just isn't practical to have to download each video one at a time and then launch it in a viewer. Also, at present, I don't think there is a way to consume the traces that are created. Those are usually shown in the HTML reporter. When support for those features is added, I think Playwright will surge forward in adoptability. As it stands right now, the reporting is still the weakest spot for Playwright with ADO.

@chris-strudel Yes, we definitely see the reason to provide dedicated playwright reporting hence we are actively working on it.

**Comment by vvs11** [comment]
Hi Folks, 

We are excited to introduce the new reporting feature for the Microsoft Playwright Testing service. With this feature, you can publish your test results and artifacts to the service and view them in the MPT portal. You'll have all the information needed to diagnose failed tests. The portal also hosts all collected artifacts, including the trace viewer. Integrating this feature with your existing test suite is straightforward and easy.

You can read more about it here: https://aka.ms/mpt/reporting-announce

Currently, the feature is invite-only preview. You can sign up using this link: https://aka.ms/mpt/reporting-signup

Thanks

**Comment by gcelia86** [comment]
Does this new feature link test cases back to ADO test plans? Currently the playwright test runs are completely separate to manual tests and all of the traceability features that come with ADO
